libgearman-server/struct/io.h

```diff
@@ -99,29 +99,34 @@ struct gearmand_io_st
     revents= 0;
   }
 
-  uint32_t created_id;
-  uint32_t created_id_next;
-  size_t send_buffer_size;
-  size_t send_data_size;
-  size_t send_data_offset;
-  size_t recv_buffer_size;
-  size_t recv_data_size;
-  size_t recv_data_offset;
-  gearmand_connection_list_st *universal;
-  gearmand_io_st *next;
-  gearmand_io_st *prev;
-  gearmand_io_st *ready_next;
-  gearmand_io_st *ready_prev;
-  gearmand_con_st *context;
-  char *send_buffer_ptr;
-  gearmand_packet_st *recv_packet;
-  char *recv_buffer_ptr;
-  gearmand_packet_st packet;
-  gearman_server_con_st *root;
+  uint32_t created_id{};
+  uint32_t created_id_next{};
+  size_t send_buffer_size{};
+  size_t send_data_size{};
+  size_t send_data_offset{};
+  size_t recv_buffer_size{};
+  size_t recv_data_size{};
+  size_t recv_data_offset{};
+  gearmand_connection_list_st *universal{nullptr};
+  gearmand_io_st *next{nullptr};
+  gearmand_io_st *prev{nullptr};
+  gearmand_io_st *ready_next{nullptr};
+  gearmand_io_st *ready_prev{nullptr};
+  gearmand_con_st *context{nullptr};
+  char *send_buffer_ptr{nullptr};
+  gearmand_packet_st *recv_packet{nullptr};
+  char *recv_buffer_ptr{nullptr};
+  gearmand_packet_st packet{};
+  gearman_server_con_st *root{nullptr};
   char send_buffer[GEARMAND_SEND_BUFFER_SIZE];
   char recv_buffer[GEARMAND_RECV_BUFFER_SIZE];
 
-  gearmand_io_st() { }
+  gearmand_io_st() {
+    this->options = {};
+    this->_state = GEARMAND_CON_UNIVERSAL_INVALID;
+    this->send_state = GEARMAND_CON_SEND_STATE_NONE;
+    this->recv_state = GEARMAND_CON_RECV_UNIVERSAL_NONE;
+  }
```

Comment:
This seems like a significant change. Can you explain?

Reply:
It was done in order to fix
```
Viva64-EM
full
124
/nfs/home/dmka/src/gearmand/libgearman-server/struct/io.h
error
V730
Not all members of a class are initialized inside the constructor. Consider inspecting: options, _state, send_state, recv_state, events, revents, ...
false
3

  gearmand_io_st() { }
------------
```
see https://github.com/p-alik/gearmand/issues/8